go_lib/updater/updater.go

```
@@ -623,8 +619,17 @@ func (u *Updater[R]) processPendingRelease(index int, release R) {
 		}
 	}
 
-	// release is predicted to be Deployed
 	if releaseRequirementsMet {
+		// check: already has predicted release and current is a patch
+		if u.predictedReleaseIndex >= 0 {
+			previousPredictedRelease := u.releases[u.predictedReleaseIndex]
+			if previousPredictedRelease.GetVersion().Major() < release.GetVersion().Major() || previousPredictedRelease.GetVersion().Minor() < release.GetVersion().Minor() {
+				_ = u.updateStatus(release, fmt.Sprintf("Awaiting for %s release to be deployed", previousPredictedRelease.GetName()), PhasePending)
```

```suggestion
				if err := u.updateStatus(release, fmt.Sprintf("Awaiting for %s release to be deployed", previousPredictedRelease.GetName()), PhasePending); err != nil {
				u.logger.Error("update status", slog.String("error", err.Error()))
				}
```

---

why we ignore error?

---

+